src/rgw/rgw_bucket.cc

```diff
@@ -264,7 +264,7 @@ void check_bad_user_bucket_mapping(rgw::sal::Store* store, rgw::sal::User* user,
         cout << "bucket info mismatch: expected " << actual_bucket << " got " << bucket << std::endl;
         if (fix) {
           cout << "fixing" << std::endl;
-	  r = actual_bucket->chown(dpp, user, nullptr, null_yield);
+	  r = actual_bucket->chown(dpp, user, null_yield);
```

Comment:
i'm having a lot of trouble making sense of this `check_bad_user_bucket_mapping()` logic. if i'm reading it right, `actual_bucket`'s owner will always be `user` because that's what we pass into `store->get_bucket()` above. so this `chown()` call will always unlink from `user`, re-link to `user`, and rewrite all of the bucket's object acls

before commit 8f4ce21ce7f6beaea76c37c72d65f7c93a0ccc2b this just called `actual_bucket->link(dpp, user, null_yield)` directly